src/crypto/muhash.cpp

```diff
@@ -299,7 +299,7 @@ Num3072 MuHash3072::ToNum3072(Span<const unsigned char> in) {
     unsigned char tmp[Num3072::BYTE_SIZE];
 
     uint256 hashed_in{(HashWriter{} << in).GetSHA256()};
-    ChaCha20(hashed_in.data(), hashed_in.size()).Keystream(tmp, Num3072::BYTE_SIZE);
+    ChaCha20(hashed_in.data(), uint256::size()).Keystream(tmp, Num3072::BYTE_SIZE);
```

Comment:
I think the pattern is generally `a.data(), a.size()`. If you want to change this, it might be best to use spans instead of the changes here in this pull.

Reply:
i am unfamiliar with spans, how would that work?

Reply:
See `std::span` and our implementation of it `Span`

Reply:
Thanks, attempted in b8f72010a424c48cabc92d8f0e7a43e8eb8460e8.
Not sure if I should add this commit in a separate PR.